xen/include/xen/iommu.h

```diff
@@ -195,6 +206,12 @@ struct iommu_ops {
     int __must_check (*iotlb_flush_all)(struct domain *d);
     int (*get_reserved_device_memory)(iommu_grdm_t *, void *);
     void (*dump_p2m_table)(struct domain *d);
+#ifdef CONFIG_HAS_COPROC
+    int (*assign_coproc)(struct domain *d, device_t *dev);
```

Comment:
I would highly recommend we use iommu_ prefix here for better readability later, e.g.
hd->platform_ops->assign_coproc
and
hd->platform_ops->iommu_assign_coproc
because in the former example w/o knowing the context you can't guess what kind of assignment is happening